doc/modules/cross_validation.rst

```diff
@@ -209,6 +209,19 @@ The following sections list utilities to generate indices
 that can be used to generate dataset splits according to different cross
 validation strategies.
 
+Cross-validation iterators for i.i.d. data.
+===========================================
+
+Data is Independent Identically Distributed (i.i.d.) if the individual samples
+of the data are independent of each other and are drawn from an identical
+distribution.
+
+Simply put, it means that each data point is created equally, and there is no
+covariance between the different data points.
```

Comment:
Maybe we should add that this hypothesis rarely strictly holds in practice: if we know
that samples have be generated using a time-dependent process, it's safer to use a
time-series aware cross-validation scheme. Similarly if we know that the generative
process has a group structure (samples from collected from different subjects,
experiments, measurement devices) it safer to use group-wise cross-validation.

And put some restructured text references to those sections in the doc.